### DISCUSSION TITLE
Not getting video packet count while go live

### DISCUSSION THREAD

**Discussion by sidharth7891** [ant-media/Ant-Media-Server#6390] [General]
When we use adaptive streaming, the video packet count is 0. If we disable adaptive streaming, we get the correct video packet count. Can you please check why the video packet count is 0 when using adaptive streaming?

Below are the steps to reproduce the issue:

1. Add the sample plugin to your Ant Media Server. Please use this link: [Sample Plugin](https://drive.google.com/file/d/1gWMvm9zBp_gEBthF3LGSFeCscVPZUW2R/view?usp=drive_link).
2. Add adaptive streaming.
3. Create a stream.
4. Register your stream by calling this API: /sample-plugin/register/{streamId}
5. After registration, onVideoPacket function is not working. 
   You can also check the packet size using this 
   API: /sample-plugin/getPacketSize/{streamId}

**Comment by burak-58** [comment]
Hi @sidharth7891,
This may not be an issue. Let me convert this to a discussion so that fasret response. If we decide that it is an issue, we can convert back.

**Reply by sumitrathi** [reply]
@burak-58 
Is their any setting to get video packet size when adaptive bitrates are enabled? without enabling adaptive bitrates it works perfectly fine.

**Reply by lastpeony** [reply]
I checked and here is the plugin behavior:

If ABR is enabled:
audio packet is coming
video packet is not coming
audio frame is coming
video frame is coming
If ABR is not enabled:
audio packet is coming
video packet is coming
audio frame is coming
video frame is not coming

Today we will discuss with tech team and try to make video pkt available if abr is enabled. @sidharth7891 @sumitrathi

**Reply by lastpeony** [reply]
We discussed this matter in detail.

Regarding the case where the video packet does not come when ABR is enabled:
We are aware of a related issue. Once this issue is resolved, we will be able to provide the video packet even when ABR is enabled. This is an advanced use case. If you inform us about your specific use case and license count, we can expedite the implementation. Please contact me directly at emre.guney@antmedia.io. Alternatively, if you are already in contact with support via email, you can inform them as well.

Regarding the case where the video frame does not come when ABR is not enabled:
It is supposed to work when application setting forceDecoding:true is set. However, I have conducted several tests and found that it also does not work. I have created a related issue for this:
https://github.com/ant-media/Ant-Media-Server/issues/6401

@sidharth7891 @sumitrathi

**Reply by sumitrathi** [reply]
@lastpeony thanks for the reply. Yes, mention the same to the support team. We are using a paid version of Antmedia and using AWS marketplace. Right now we are using 1 license but if we solve this issue we may reach 10 licenses. Here is our use case, we would like to calculate the total stream sent by the customer and based on stream we are planning to charge our customer like 10 GB stream has a different plan, 100 GB stream different plan. We are open if there is any alternative way to achieve this functionality. Thanks for your suggestions.
